core/templates/pages/learner-dashboard-page/carousel-bar.component.spec.ts

```diff
@@ -0,0 +1,331 @@
+
+    component.scroll(false);
+
+    expect(component.carouselScrollPositionPx).toBe(expectedScrollPositionPx);
+    expect(carouselSelector.scrollBy).toHaveBeenCalledWith({
+      top: 0,
+      left: expectedScrollPositionPx,
+      behavior: 'smooth',
+    });
+
+    expect(component.disableLeftButton).toBe(false);
+    expect(component.disableRightButton).toBe(false);
+  });
+
+  it('should scroll untracked topics to left when card is at right end', () => {
+    const carouselSelector = document.createElement('div');
+    spyOn(document, 'querySelector').and.returnValue(carouselSelector);
+
+    carouselSelector.scrollTo = jasmine.createSpy('scrollTo');
+    Object.defineProperty(carouselSelector, 'scrollWidth', {
+      configurable: true,
+      value: 500,
+    });
+
+    Object.defineProperty(carouselSelector, 'clientWidth', {
+      configurable: true,
+      value: 50,
+    });
+    component.scrollUntrackedTopics = true;
+    component.carouselScrollPositionPx = 460;
+
+    const direction = -1;
+
+    const expectedScrollPositionPx =
+     component.carouselScrollPositionPx +
+     (direction * AppConstants.UNTRACKED_TILE_SWAP_WIDTH_PX);
+
+    spyOn(carouselSelector, 'scrollBy');
+
+    component.scroll(true);
+
+    expect(component.carouselScrollPositionPx).toBe(expectedScrollPositionPx);
+    expect(carouselSelector.scrollBy).toHaveBeenCalledWith({
+      top: 0,
+      left: (direction * AppConstants.UNTRACKED_TILE_SWAP_WIDTH_PX),
+      behavior: 'smooth',
+    });
+
+    expect(component.disableLeftButton).toBe(false);
+    expect(component.disableRightButton).toBe(false);
+  });
+
+
+  it('should scroll the topic to right if topics are not untracked', () => {
+    const carouselSelector = document.createElement('div');
+    spyOn(document, 'querySelector').and.returnValue(carouselSelector);
+
+    carouselSelector.scrollTo = jasmine.createSpy('scrollTo');
+    Object.defineProperty(carouselSelector, 'scrollWidth', {
+      configurable: true,
+      value: 500,
+    });
+
+    Object.defineProperty(carouselSelector, 'clientWidth', {
+      configurable: true,
+      value: 50,
+    });
+
+    component.scrollUntrackedTopics = false;
+    component.carouselScrollPositionPx = 0;
+
+    const direction = 1;
+
+    const expectedScrollPositionPx =
+     component.carouselScrollPositionPx +
+     (direction * AppConstants.LEARNER_DASHBOARD_TILE_WIDTH_PX);
+
+    spyOn(carouselSelector, 'scrollBy');
+
+    component.scroll(false);
+
+    expect(component.carouselScrollPositionPx).toBe(expectedScrollPositionPx);
+    expect(carouselSelector.scrollBy).toHaveBeenCalledWith({
+      top: 0,
+      left: (direction * AppConstants.LEARNER_DASHBOARD_TILE_WIDTH_PX),
+      behavior: 'smooth',
+    });
+
+    expect(component.disableLeftButton).toBe(false);
+    expect(component.disableRightButton).toBe(false);
+  });
+
+  it('should disable left button' +
+   ' when carouselScrollPositionPx is less than or equal to 0', () => {
```

Comment:
What does this test ensure for the user? The test description should not discuss implementation specifics. Please see https://github.com/oppia/oppia/wiki/Frontend-tests#name-tests-clearly

Reply:
Ditto below.